### PR TITLE
Feature/review tr styles

### PR DESCRIPTION
**Pull request by NicolaSaunders**
Proposed solution for https://github.com/w3c/w3c-website/issues/207

- Adjust styles to tighten up vertical spacing of the whole TR listing page
- Make TR listing easier to visually scan
    - Remove number from contextual text, as it seems of minimal benefit to people
    - Visually hide contextual text in TR translations `<dt>`, leaving it available for screen reader users
    - Add CSS to compensate for additional inline space created by the visually hidden text before the css-inserted colon at the end of the `<dt>`

### CODE REVIEW

File: design-system-templates/components/tr-search-results-family.html.twig [hunk 1]
```diff
@@ -55,7 +55,7 @@
                 <dd>Michael Cooper</dd>
             </div>
             <div class="translation-list">
-                <dt>2 translations for Accessible Rich Internet Applications (WAI-ARIA) 1.0</dt>
+                <dt>Translations <span class="visuallyhidden">/span> Accessible Rich Internet Applications (WAI-ARIA) 1.0</span></dt>
```
Comment (vivienlacourba) [inline design-system-templates/components/tr-search-results-family.html.twig:58]:
```suggestion
                <dt>Translations <span class="visuallyhidden">for Accessible Rich Internet Applications (WAI-ARIA) 1.0</span></dt>
```

**Review by vivienlacourba — CHANGES_REQUESTED** [review]

File: design-system-templates/components/tr-search-results-family.html.twig (hunk 1 above)
Comment (NicolaSaunders) [reply design-system-templates/components/tr-search-results-family.html.twig]: Apologies @vivienlacourba - I have pushed an amendment for that.

File: assets-src/styles/sass/80-templates/_technical-reports.scss [hunk 2]
```diff
@@ -31,14 +31,20 @@
...
 }
 
 .tr-list {
 	.tr-list__item {
 		@include stack(0.5em);
 		@include txt-pluto;
-		padding-bottom: rem(24);
-		padding-top: rem(24);
+        border-block-end: 1px solid $twiki-gray;
+		padding-block-end: rem(8);
```
Comment (deniak) [inline assets-src/styles/sass/80-templates/_technical-reports.scss:46]: Fix indentation
```suggestion
		border-block-end: 1px solid $twiki-gray;
		padding-block-end: rem(8);
```

File: assets-src/styles/sass/80-templates/_technical-reports.scss [hunk 3]
```diff
@@ -49,21 +55,17 @@
...
-			margin-bottom: 0;
-			margin-top: rem(5);
+			margin-block-end: 0;
+			margin-block-start: rem(5);
 		}
 
 		h3 + * {
-			margin-top: 0;
+            margin-block-start: 0;
 			order: -1;
```
Comment (deniak) [inline assets-src/styles/sass/80-templates/_technical-reports.scss:69]: Fix indentation
```suggestion
			margin-block-start: 0;
			order: -1;
```

**Review by deniak — APPROVED** [review]
I made 2 suggestions to fix the indentation but otherwise, LGTM

**Pull request merged**